src/world/Spell/Spell_ClassScripts.cpp

```diff
@@ -81,9 +81,9 @@ class FireNova : public Spell
     static Spell* Create(Object* Caster, SpellInfo *info, bool triggered, Aura* aur) { return new FireNova(Caster, info, triggered, aur); }
 
     bool HasFireTotem = false;
-    uint8 CanCast(bool tolerate)
+    SpellCastResult canCast(bool tolerate)
     {
-        uint8 result = Spell::CanCast(tolerate);
+        SpellCastResult result = Spell::canCast(tolerate);
```

Comment:
Is explicit typing required here? If not, please use implicit typing.

Reply:
Not required, but canCast returns value in SpellCastResult and not uint8.

Reply:
@Appled but what I mean is you should only be specifying the type if auto is evaluating to the wrong type, and I don't see why it would be here.